pkg/api/v1alpha1/users.go

```diff
@@ -89,6 +94,41 @@ func (r *Router) listUsers(c *gin.Context) {
 			for _, v := range convertedVals {
 				queryMods = append(queryMods, qm.Or("LOWER(email) = LOWER(?)", v))
 			}
+		case "metadata":
+			// metadata should be a JSON formatted object used to filter users with
+			// specific metadata values
+			//
+			// this object should be in the format of:
+			// ?metadata=key1=value&metadata=path.to.nested.field=value2
+			//
+			const KVPartsLen = 2
+
+			for _, searchString := range val {
+				searchKV := strings.SplitN(searchString, "=", KVPartsLen)
+				if len(searchKV) < KVPartsLen {
+					r.Logger.Error("invalid metadata query format", zap.String("metadata", searchString))
+					sendError(c, http.StatusBadRequest, "invalid metadata query format: "+searchString)
+
+					return
+				}
+
+				searchKey, searchValue := searchKV[0], searchKV[1]
+
+				pathComponents := strings.Split(searchKey, ".")
+				sqlPath := fmt.Sprintf("{%s}", strings.Join(pathComponents, ","))
```

Comment:
I think we should probably have some validation on the key here. Maybe only allow alpha numeric characters? The same validation should be added when posting metadata as well.